spack/package.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Arbor(CMakePackage, CudaPackage):
+    """Arbor is a high-performance library for computational neuroscience
+    simulations."""
+
+    homepage = 'https://github.com/arbor-sim/arbor/'
+    git      = 'https://github.com/arbor-sim/arbor.git'
+    url      = 'https://arbor.readthedocs.io/en/latest/'
+    maintainers = ['bcumming', 'halfflat']
+
+    version('master', branch='master', submodules=True)
+    version('0.5', 'd0c8a4c7f97565d7c30493c66249be794d1dc424de266fc79cecbbf0e313df59', url='https://github.com/arbor-sim/arbor/releases/download/v0.5/arbor-v0.5-full.tar.gz')
+
+    variant('assertions', default=False, description='Enable arb_assert() assertions in code.')
+    variant('doc', default=False, description='Build documentation.')
+    variant('mpi', default=False, description='Enable MPI support')
+    variant('neuroml', default=True, description='Build NeuroML support library.')
+    variant('python', default=True, description='Enable Python frontend support')
+    variant('vectorize', default=False, description='Enable vectorization of computational kernels')
+
+    # https://arbor.readthedocs.io/en/latest/install/build_install.html?highlight=requirements#compilers
+    conflicts('%gcc@:8.3')
+    conflicts('%clang@:8.0')
```

Comment:
We list clang 8 as min version. Why would it conflict? Because of its stdlib?

Reply:
I just copied over our current configuration.
It is true that `:8.0` means `<=8`, so it should be `:7`.

The spack syntax of using only open intervals for specifying version ranges is awkward.

Reply:
I will fix this.

Reply:
There's an open PR for this, but not there yet: https://github.com/spack/spack/pull/20258